src/Misc/XMLwrapper.cpp

```diff
@@ -670,15 +670,26 @@ void XMLwrapper::add(const XmlNode &node_)
 std::vector<XmlNode> XMLwrapper::getBranch(void) const
 {
     std::vector<XmlNode> res;
-    mxml_node_t *current = node->child;
+    mxml_node_t *current = mxmlGetFirstChild(node);
     while(current) {
-        if(current->type == MXML_ELEMENT) {
+        if (mxmlGetType(current) == MXML_ELEMENT) {
+        #if MXML_MAJOR_VERSION == 3
```

Comment:
After double checking with my system version of mxml (a rather old one). It appears that not all versions have MXML_MAJOR_VERSION defined. For particularly old versions of mxml, can you use the new code if MXML_MAJOR_VERSION is defined and is equal to 3?

Also, by convention \#if \#else and \#endif are placed with no indention.